### PR TITLE
Refactor some things

### PR DESCRIPTION
**Pull request by QuentinBisson**
Refactor a few things to simplify https://github.com/giantswarm/prometheus-meta-operator/pull/1192/files

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [X] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: flag/service/provider/provider.go [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package provider
 
 type Provider struct {
-	Kind string
+	Kind   string
+	Flavor string
```
Comment (QuentinBisson) [inline flag/service/provider/provider.go:5]: Add flavor property

File: flag/service/service.go [hunk 2]
```diff
@@ -16,14 +16,14 @@ import (
 
 // Service is an intermediate data structure for command line configuration flags.
 type Service struct {
+	Grafana      grafana.Grafana
```
Comment (QuentinBisson) [inline flag/service/service.go:19]: Sort correctly

File: helm/prometheus-meta-operator/templates/alertmanager/alertmanager-ingress.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ metadata:
     {{- end }}
   labels:
     {{- include "labels.common" . | nindent 4 }}
-  name: {{ include "resource.default.name" . }}-alertmanager
+  name: alertmanager
```
Comment (QuentinBisson) [inline helm/prometheus-meta-operator/templates/alertmanager/alertmanager-ingress.yaml:17]: The others are all called alertmanager

File: helm/prometheus-meta-operator/templates/alertmanager/alertmanager.yaml
[the remark this answers is not in the document]
Comment (marieroque) [reply helm/prometheus-meta-operator/templates/alertmanager/alertmanager.yaml:1]: Apart this comment, LGTM :-)

**Review by marieroque — APPROVED** [review]

**Pull request merged**